app/src/main/java/com/zulip/android/networking/ZulipAsyncPushTask.java

```diff
@@ -93,11 +96,14 @@ public void setProperty(String key, String value) {
 
     protected String doInBackground(String... api_path) {
         try {
-            return request.execute(api_path[0], api_path[1]);
-        } catch (HttpResponseException e) {
-            handleHTTPError(e);
+            Response response = request.execute(api_path[0], api_path[1]);
+            return response.body().string();
```

Comment:
this might be a good place to check for http error status codes being returned, and calling `handleHTTPError`.